mos_tests/rabbitmq_oslo/test_restarts.py

```diff
@@ -416,3 +443,84 @@ def test_load_messages_and_restart_prim_nonprim_ctrlr(restart_ctrlr, env):
+    5. To be able to use port 5000 from any node open it in IPTables;
+    6. Send GET curl request from any node to 'oslo_msg_check_client'
+        located on compute node and check that response will '200';
+    7. Restart RabbitMQ-server on one OR all controller(s) one-by-one;
+    8. Wait until RabbitMQ service will be up and cluster synchronised;
+    9. Send GET curl request from any node to 'oslo_msg_check_client'
+        located on compute node and check that response will '200';
+    10. Remove rule from step (5) from IPTables.
+    """
+    exp_resp = 200   # expected response code from curl from RPC client
+    timeout_min = 2  # (minutes) time to wait for RPC server/client start
+
+    all_nodes = env.get_all_nodes()
+    controllers = env.get_nodes_by_role('controller')
+    controller = random.choice(controllers)
+    compute = random.choice(env.get_nodes_by_role('compute'))
+
+    # Get management IPs of all controllers
+    ctrl_ips = get_mngmnt_ip_of_ctrllrs(env)
+
+    # Install and configure tool on controller and compute
+    for node in (controller, compute):
+        with node.ssh() as remote:
+            kwargs = vars_config(remote)
+            if 'controller' in node.data['roles']:
+                # wait when rabbit will be ok after snapshot revert
+                wait_for_rabbit_running_nodes(remote, len(controllers))
+            install_oslomessagingchecktool(remote, **kwargs)
+            configure_oslomessagingchecktool(
+                remote, ctrl_ips, kwargs['nova_user'], kwargs['nova_pass'],
+                kwargs['cfg_file_path'], kwargs['sample_cfg_file_path'])
+
+    # client: run 'oslo_msg_check_client' on compute
+    with compute.ssh() as remote:
+        rpc_client_ip = rabbit_rpc_client_start(
+            remote, kwargs['cfg_file_path'])
+    # server: run 'oslo_msg_check_server' on controller
+    with controller.ssh() as remote:
+        rabbit_rpc_server_start(remote, kwargs['cfg_file_path'])
+
+    # node -> client: Check GET before controller(s) restart
+    node = random.choice(all_nodes)
+    with node.ssh() as remote:
+        logger.debug('GET: [{0}] -> [{1}]'.format(remote.host, rpc_client_ip))
+        # need to wait for server/client start
+        wait(lambda: get_http_code(rpc_client_ip) == exp_resp,
+             timeout_seconds=60 * timeout_min,
+             sleep_seconds=20,
+             waiting_for='RPC server/client to start')
+
+    # Restart RabbinMQ server on one/all controller(s)
+    with controller.ssh() as remote:
+        if restart_ctrlr == 'one':
+            restart_rabbitmq_serv(env, remote=remote)
+        elif restart_ctrlr == 'all':
+            restart_rabbitmq_serv(env, remote=None)
+
+    # node -> client: Check GET after controller(s) restart
+    node = random.choice(all_nodes)
+    with node.ssh() as remote:
+        logger.debug('GET: [{0}] -> [{1}]'.format(remote.host, rpc_client_ip))
+        assert get_http_code(rpc_client_ip) == exp_resp
```

Comment:
He uses it here:
logger.debug('GET: [{0}] -> [{1}]'.format(remote.host, rpc_client_ip))
But we can get host (hostname) directly from node.